### PR TITLE
[SYCL] [NATIVECPU] Disabling default `-sycl-opt` for NativeCPU and updating docs

### PR DESCRIPTION
**Pull request by uwedolinsky**
This PR no longer passes `-mllvm -sycl-opt` to the device compiler invocation for NativeCPU to enable the original full llvm optimization pipeline.

`EnvironmentVariables.md` has been updated to document NativeCPU and its usage in environment variables.

### CODE REVIEW

**Review by steffenlarsen — APPROVED** [review]
Docs changes LGTM! Maybe I just don't see the relation, but could this have been two PRs, i.e. one for `-sycl-opt` and one for the docs changes? If yes, we should aim for having that in the future. It makes the history cleaner and the discussion in the PR can be kept focused.

File: clang/test/Driver/sycl-native-cpu.cpp [hunk 1]
```diff
@@ -17,3 +17,11 @@
...
 // CHECK-WIN-DAG: {{.*}}"-fsycl-is-host"{{.*}}"-gcodeview"
 // CHECK-WIN-NOT: dwarf
+
+// checks that -sycl-opt is not enabled by default on NativeCPU so that the full llvm optimization is enabled
+// RUN:   %clang -fsycl -fsycl-targets=native_cpu -### %s 2>&1 | FileCheck -check-prefix=CHECK-OPTS %s
+// CHECK-OPTS-NOT: -sycl-opt
+
+// checking that explicitly enabling -sycl-opt still works
+// RUN:   %clang -fsycl -fsycl-targets=native_cpu -mllvm -sycl-opt -### %s 2>&1 | FileCheck -check-prefix=CHECK-OPT_ON %s
+// CHECK-OPT_ON: -sycl-opt
```
Comment (bader) [inline clang/test/Driver/sycl-native-cpu.cpp:27]: This check doesn't make sense to me. We already have tests checking testing `-mllvm` option and I don't think this test adds any value.

Comment (uwedolinsky) [reply clang/test/Driver/sycl-native-cpu.cpp]: I removed the check. Thank you.

**Comment by uwedolinsky** [comment]
> Docs changes LGTM! Maybe I just don't see the relation, but could this have been two PRs, i.e. one for `-sycl-opt` and one for the docs changes? If yes, we should aim for having that in the future. It makes the history cleaner and the discussion in the PR can be kept focused.

Yes, sorry, these could have been separate PRs, and I'll bare that in mind. Thank you.

**Review by mdtoguchi — APPROVED** [review]
OK for driver

**Review by PietroGhg — APPROVED** [review]
LGTM, thank you

**Comment by uwedolinsky** [comment]
Hello @intel/llvm-gatekeepers , this PR looks ready to merge. Thank you!

**Pull request merged**